spec/p2p/reactor/p2p-api.md

```diff
@@ -0,0 +1,276 @@
+# API for Reactors
+
+This document describes the API provided by the p2p layer to the protocol
+layer, namely to the registered reactors.
+
+This API consists of two interfaces: the one provided by the `Switch` instance,
+and the ones provides by multiple `Peer` instances, one per connected peer.
+The `Switch` instance is provided to every reactor as part of the reactor's
+[registration procedure][reactor-registration].
+The multiple `Peer` instances are provided to every registered reactor whenever
+a [new connection with a peer][reactor-addpeer] is established.
+
+
+## `Switch` API
+
+The [`Switch`][switch-type] is the central component of the p2p layer
+implementation.  It manages all the reactors running in a node and keeps track
+of the connections with peers.
+The table below summarizes the interaction of the standard reactors with the `Switch`:
+
+| `Switch` API method                                     | consensus | block sync | state sync | mempool | evidence  | PEX   |
+|--------------------------------------------|-----------|------------|------------|---------|-----------|-------|
+| `Peers() IPeerSet`                         | x         | x          |            |         |           | x     |
+| `NumPeers() (int, int, int)`               |           | x          |            |         |           | x     |
+| `Broadcast(Envelope) chan bool`            | x         | x          | x          |         |           |       |
+| `MarkPeerAsGood(Peer)`                     | x         |            |            |         |           |       |
+| `StopPeerForError(Peer, interface{})`      | x         | x          | x          | x       | x         | x     |
+| `StopPeerGracefully(Peer)`                 |           |            |            |         |           | x     |
+| `Reactor(string) Reactor`                  |           | x          |            |         |           |       |
+
+The above list is not exhaustive as it does not include all the `Switch` methods
+invoked by the PEX reactor, a special component that should be considered part
+of the p2p layer. This document does not cover the operation of the PEX reactor
+as a connection manager.
+
+### Peers State
+
+The first two methods in the switch API allow reactors to query the state of
+the p2p layer: the set of connected peers.
+
+    func (sw *Switch) Peers() IPeerSet
+
```

Comment:
```suggestion
   ```go
    func (sw *Switch) Peers() IPeerSet
   ```
```

Reply:
Is the rendering odd here? It looks the same as in the other uses of this code block.

Reply:
The point is that I would need to change this thorough the document, maybe also in the other file.

Do you think that having specific Go-code highlighting renders it easier to read?

Reply:
I do, but not too much for such small blocks.
Update only if you get free cycles.